core/trino-main/src/main/java/io/trino/execution/querystats/QueryRuleStats.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.execution.querystats;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import io.airlift.stats.TDigest;
+import io.trino.plugin.base.metrics.TDigestHistogram;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+public class QueryRuleStats
+{
+    private final String rule;
+    private long invocations;
+    private long hits;
+    private final TDigest time;
```

Comment:
do we need `time. digest` property? it takes a lot of space and it will grow query json that is too big already.

Reply:
The number of rules to be presented in JSON is configurable, we don't present all rules statistics in response.

Reply:
`time` is serialized as `TDigestHistogram` as it could be read from JSON. `TimeDistribution` and `TimeDistributionSnapshot` would fail `TestQueryStats`